nvflare/fuel/f3/cellnet/cell.py

```diff
@@ -1791,7 +1791,11 @@ def _process_received_msg(self, endpoint: Endpoint, connection: Connection, mess
                 elif msg_type == MessageType.REQ:
                     # see whether we can offer a listener
                     allow_adhoc = self.connector_manager.is_adhoc_allowed(oi, self.my_info)
```

Comment:
A very tricky issue is to how to avoid the situation that both cell A and cell B are offering listeners to each other at the same time (which happens a lot when we did the stress testing). We don't want this to happen since it would create two possible paths between two cells. Another factor is that server side is usually more willing to offer listeners since it is usually a neutral site.

Given these considerations, the rule for creating conn listener is as follows:
1. If both cells are on the server side, no ad-hoc conns between them;
2. If one cell is on client, another is on server, then server-side cell will offer;
3. If both cells are client side, then only the one with a greater cell name (strings compared) will offer.

The rules have always been like this, but we didn't implement Rule #2 correctly. This PR fixes it.